pkg/agent/templates/templates_windows.go

```diff
@@ -0,0 +1,174 @@
+          runtime_root = ""
+          privileged_without_host_devices = false
+          base_runtime_spec = ""
+    [plugins."io.containerd.grpc.v1.cri".cni]
+      bin_dir = "{{ replace .NodeConfig.AgentConfig.CNIBinDir }}"
+      conf_dir = "{{ replace .NodeConfig.AgentConfig.CNIConfDir }}"
+      max_conf_num = 1
+      conf_template = ""
+    [plugins."io.containerd.grpc.v1.cri".registry]
+      config_path = ""
+      {{range $k, $v := .PrivateRegistryConfig.Configs }}
+      [plugins."io.containerd.grpc.v1.cri".registry.auths]
+      {{ if $v.Auth }}
+        [plugins."io.containerd.grpc.v1.cri".registry.configs.auth."{{$k}}"]
+          {{ if $v.Auth.Username }}username = {{ printf "%q" $v.Auth.Username }}{{end}}
+          {{ if $v.Auth.Password }}password = {{ printf "%q" $v.Auth.Password }}{{end}}
+          {{ if $v.Auth.Auth }}auth = {{ printf "%q" $v.Auth.Auth }}{{end}}
+          {{ if $v.Auth.IdentityToken }}identitytoken = {{ printf "%q" $v.Auth.IdentityToken }}{{end}}
+      {{end}}
+      [plugins."io.containerd.grpc.v1.cri".registry.configs]
+      {{ if $v.TLS }}
+        [plugins."io.containerd.grpc.v1.cri".registry.configs.tls."{{$k}}"]
+          {{ if $v.TLS.CAFile }}ca_file = "{{ $v.TLS.CAFile }}"{{end}}
+          {{ if $v.TLS.CertFile }}cert_file = "{{ $v.TLS.CertFile }}"{{end}}
+          {{ if $v.TLS.KeyFile }}key_file = "{{ $v.TLS.KeyFile }}"{{end}}
+          {{ if $v.TLS.InsecureSkipVerify }}insecure_skip_verify = true{{end}}
+      {{end}}
+      {{end}}
+      [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
+      {{ if .PrivateRegistryConfig.Mirrors }}
+      {{range $k, $v := .PrivateRegistryConfig.Mirrors }}
+        [plugins."io.containerd.grpc.v1.cri".registry.mirrors."{{$k}}"]
+	        endpoint = [{{range $i, $j := $v.Endpoints}}{{if $i}}, {{end}}{{printf "%q" .}}{{end}}]
+	    {{if $v.Rewrites}}
+	      [plugins."io.containerd.grpc.v1.cri".registry.mirrors."{{$k}}".rewrite]
+	      {{range $pattern, $replace := $v.Rewrites}}
+	        "{{$pattern}}" = "{{$replace}}"
+	      {{end}}
+	    {{end}}
+      {{end}}
+    {{end}}
+    [plugins."io.containerd.grpc.v1.cri".image_decryption]
+      key_model = ""
+    [plugins."io.containerd.grpc.v1.cri".x509_key_pair_streaming]
+      tls_cert_file = ""
+      tls_key_file = ""
+  [plugins."io.containerd.internal.v1.opt"]
+    path = "{{ replace .NodeConfig.Containerd.Opt }}"
+  [plugins."io.containerd.internal.v1.restart"]
+    interval = "10s"
+  [plugins."io.containerd.metadata.v1.bolt"]
+    content_sharing_policy = "shared"
+  [plugins."io.containerd.runtime.v2.task"]
+    platforms = ["windows/amd64", "linux/amd64"]
+  [plugins."io.containerd.service.v1.diff-service"]
+    default = ["windows", "windows-lcow"]
+`
+
+func ParseTemplateFromConfig(templateBuffer string, config interface{}) (string, error) {
+	out := new(bytes.Buffer)
+	funcs := template.FuncMap{
+		"replace": func(s string) string {
+			return strings.Replace(s, "\\", "\\\\", -1)
```

Comment:
this appears to just be used to escape double backslashes in paths, is that correct?
```suggestion
			return strings.ReplaceAll(s, "\\", "\\\\")
```